sw/device/lib/msg_print.h

```diff
@@ -0,0 +1,53 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef _MSG_PRINT_H_
+#define _MSG_PRINT_H_
+
+// Pointer to a function that prints a character.
+typedef void (*print_char_func)(char);
+
+// Identifiers for string format type specifiers.
+static const char kFormatSpecifier = '%';
+static const char kBinary = 'b';       // Print binary [SystemVerilog style].
+static const char kDecimalI = 'i';     // Signed decimal.
+static const char kDecimal = 'd';      // Signed decimal.
+static const char kUnsigned = 'u';     // Unsigned decimal.
+static const char kOctal = 'o';        // Octal.
+static const char kAsciiChar = 'c';    // Single character byte.
+static const char kAsciiStr = 's';     // This is currently not supported.
+static const char kHexUpper = 'X';     // Upper case hex.
+static const char kHexAltUpper = 'H';  // Upper case hex [SystemVerilog style].
+static const char kHexLower = 'x';     // Lower case hex.
+static const char kHexAltLower = 'h';  // Lower case hex [SystemVerilog style].
+static const char kPercent = '%';      // Escape '%'.
```

Comment:
This is going to duplicate these symbols in every translation unit that includes this file. The correct way to export a constant in C is
```c
// foo.h
extern const char kMyConst;
// foo.c
const char kMyConst = ...;
```

Reply:
Fixed.